### PR TITLE
fixed: reveive_tokenized_share handler #LSC-337

### PR DESCRIPTION
**Pull request by swelf19**
its alive!

I had to fix MockApi. Mocked add_humanize differs from  production one, and method fails in the unit tests when building Addr for a share_module

### CODE REVIEW

**Comment by zavgorodnii** [comment]
yaaaaaay
@ratik can you test it with your testkit?

File: contracts/lido_cosmos_hub/src/tokenized.rs [hunk 1]
```diff
@@ -152,6 +178,18 @@ fn get_largest_validator(
     }))
 }
 
+pub fn get_module_address(deps: Deps, module_name: String) -> StdResult<String> {
+    // https://github.com/iqlusioninc/liquidity-staking-module/blob/faf413d4624af0a6fa8aac2c359d1b1b4f6adbc9/x/staking/types/tokenize_share_record.go#L8
+    let mut hasher = Sha256::new();
+    hasher.update(module_name);
+    let result = hasher.finalize();
+    // https://github.com/tendermint/tendermint/blob/master/crypto/crypto.go#L19
+    let s = result[..20].to_vec();
```
Comment (zavgorodnii) [inline contracts/lido_cosmos_hub/src/tokenized.rs]: please move 20 to a constant

File: contracts/lido_cosmos_hub/src/tokenized.rs [hunk 2]
```diff
@@ -152,6 +178,18 @@ fn get_largest_validator(
     }))
 }
 
+pub fn get_module_address(deps: Deps, module_name: String) -> StdResult<String> {
+    // https://github.com/iqlusioninc/liquidity-staking-module/blob/faf413d4624af0a6fa8aac2c359d1b1b4f6adbc9/x/staking/types/tokenize_share_record.go#L8
+    let mut hasher = Sha256::new();
+    hasher.update(module_name);
```
Comment (zavgorodnii) [inline contracts/lido_cosmos_hub/src/tokenized.rs]: this code is scary af. i don't see any way to avoid it though...

Comment (zavgorodnii) [reply contracts/lido_cosmos_hub/src/tokenized.rs]: looks like guys from cosmwasm use a shortcut: https://github.com/CosmWasm/cosmwasm/blob/e4fd6413e8a54ae4bb9ff2a1e60ef551d03f2927/packages/crypto/benches/main.rs#L81

File: contracts/lido_cosmos_hub/src/testing/mock_querier.rs [hunk 3]
```diff
@@ -113,15 +114,15 @@ impl WasmMockQuerier {
                         .iter()
                         .find(|r| r.share_token_denom == request.denom);
                     if record.is_none() {
-                        return SystemResult::Ok(ContractResult::from(to_binary(&Binary::from(
-                            response.write_to_bytes().unwrap(),
-                        ))));
+                        return SystemResult::Ok(ContractResult::from(Binary::from_base64(
+                            &Binary::from(response.write_to_bytes().unwrap()).to_base64(),
+                        )));
```
Comment (foxpy) [inline contracts/lido_cosmos_hub/src/testing/mock_querier.rs:119]: That's hell on Earth. Can't we do it in a simpler way? I haven't found a decent solution :(

Comment (foxpy) [reply contracts/lido_cosmos_hub/src/testing/mock_querier.rs:119]: `return SystemResult::Ok(ContractResult::Ok(Binary::from(response.write_to_bytes().unwrap())));` :sunglasses:

**Review by foxpy — CHANGES_REQUESTED** [review]

**Review by foxpy — APPROVED** [review]

**Review by zavgorodnii — APPROVED** [review]

**Review by ratik — APPROVED** [review]

**Pull request merged**